sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -437,4 +446,37 @@ public Map.Entry<String, String> next() {
             return new AbstractMap.SimpleImmutableEntry<>(key, value);
         }
     }
+
+    public static <T> T submitThreadPool(Supplier<T> operation, Duration timeout) {
```

Comment:
It'd be nice to update this to take a `ClientLogger` parameter and have it handle the try/catch for `RuntimeException` done in every synchronous call. The `LOGGER` would still handle the interrupt, execution, and timeout exceptions.